src/main/java/com/doeiqts/pizzavoter/domain/Pizza.java

```diff
@@ -23,20 +29,29 @@ public Pizza(Size size, Crust crust, Sauce sauce) {
         this.size = size;
         this.sauce = sauce;
         toppings.put(Topping.CHEESE, Position.ALL);
+        //it's effectively 0 but cheese is on everything...not sure how I feel about this...
+        toppingCount = 1.0;
     }
 
     public Position addTopping(Topping topping, Position position) {
-        return this.toppings.put(topping,position);
+        double old = 0.0; //if duplicate topping positions are sent need to subtract existing topping value)
```

Comment:
I think we can clean up this logic by using the return value we get from Map.put() and checking the toppingCount against the LIMIT after we put the new value and just revert it if needed. IDK, not going to hold up the pull request, just commenting for thought later.